be/src/exprs/array_functions.cpp

```diff
@@ -1199,4 +1206,213 @@ StatusOr<ColumnPtr> ArrayFunctions::array_slice(FunctionContext* ctx, const Colu
     }
 }
 
+// unpack array column, return: null_column, element_column, offset_column
+static inline std::tuple<NullColumnPtr, Column*, const UInt32Column*> unpack_array_column(const ColumnPtr& input) {
+    NullColumnPtr array_null = nullptr;
+    ArrayColumn* array_col = nullptr;
+
+    auto array = ColumnHelper::unpack_and_duplicate_const_column(input->size(), input);
+    if (array->is_nullable()) {
+        auto nullable = down_cast<NullableColumn*>(array.get());
+        array_col = down_cast<ArrayColumn*>(nullable->data_column().get());
+        array_null = NullColumn::create(*nullable->null_column());
+    } else {
+        array_null = NullColumn::create();
+        array_null->resize(array->size());
+        array_col = down_cast<ArrayColumn*>(array.get());
+    }
+
+    const UInt32Column* offsets = &array_col->offsets();
+    Column* elements = array_col->elements_column().get();
+
+    return {array_null, elements, offsets};
+}
+
+StatusOr<ColumnPtr> ArrayFunctions::array_distinct_any_type(FunctionContext* ctx, const Columns& columns) {
+    DCHECK_EQ(1, columns.size());
+    RETURN_IF_COLUMNS_ONLY_NULL(columns);
+
+    auto [array_null, elements, offsets] = unpack_array_column(columns[0]);
```

Comment:
optimized for const column? which just need compute only one array

Reply:
Const column optimize has nothing to do with unpack, should let the row size is 1 if you want to optimize const column

Reply:
as `unpack_and_duplicate_const_column` in `unpack_array_column` duplicated const columns to input->size(), whichi is unecessary. only distinct the first row is ok.

Reply:
find, I know, optimize const column is need other branch, we can optmizer it later. BTW, based on our current situation, the constant column is basically meaningless. We will unfold the constant column in many places, except  some simple expressions with branch optimization in longlong ago